modules/core/include/opencv2/core/traits.hpp

```diff
@@ -49,11 +49,23 @@
 namespace cv
 {
 
+//#define OPENCV_TRAITS_ENABLE_DEPRECATED
+
+// Hack for CUDA modules (refactoring is required for CUDA modules)
+#ifndef OPENCV_TRAITS_ENABLE_DEPRECATED
+# include "opencv2/opencv_modules.hpp"
+# ifdef HAVE_OPENCV_CUDAFEATURES2D
+#   define OPENCV_TRAITS_ENABLE_DEPRECATED
```

Comment:
Is it possible to turn this on via cmake?

Reply:
via CMake we should turn on this in two places at least:
- OpenCV compilation itself
- user applications which use OpenCV

Reply:
@alalek , sure. Maybe something like the following lines will work in both cases:
```.cmake
if (HAVE_CUDA)
    target_compile_definitions(opencv_core PUBLIC OPENCV_TRAITS_ENABLE_DEPRECATED)
endif()
```

Reply:
Replaced